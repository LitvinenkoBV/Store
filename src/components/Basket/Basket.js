import "./Basket.scss";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";

const Basket = () => {
  const arrProductsToBuy = useSelector((state) => state.productsToBuy);
  return (
    <Link to="/basket" className="header__basket">
      <img src="./img/shopping-cart.png" alt="" />
      {arrProductsToBuy.length}
    </Link>
  );
};

Basket.propTypes = {
  arrProductsToBuy: PropTypes.array,
};

export default Basket;
