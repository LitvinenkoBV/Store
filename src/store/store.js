import { createStore, combineReducers, compose, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { productsReducer } from "./products/reducer";
import { getFavoriteProductsReducer } from "./favorite_products//reducer";
import { productsToBuyReducer } from "./products_to_buy/reducer";
import { modalReducer } from "./modal/reducer";

export const rootReducer = combineReducers({
  products: productsReducer,
  favoriteProducts: getFavoriteProductsReducer,
  productsToBuy: productsToBuyReducer,
  modal: modalReducer,
});

const devTools = window.__REDUX_DEVTOOLS_EXTENSION__
  ? window.__REDUX_DEVTOOLS_EXTENSION__()
  : (f) => f;

export const store = createStore(
  rootReducer,
  compose(applyMiddleware(thunk), devTools)
);
