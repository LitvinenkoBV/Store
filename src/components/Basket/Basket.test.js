import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { legacy_createStore as createStore } from "redux";
import Basket from "./Basket";
import { rootReducer } from "../../store/store";
import { BrowserRouter } from "react-router-dom";

const store = createStore(rootReducer, {
  products: [
    { id: 1, product: "phone" },
    { id: 2, product: "laptop" },
  ],
});

describe("<Basket/> component", () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Basket />
      </BrowserRouter>
    </Provider>
  );
  it("should render", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
