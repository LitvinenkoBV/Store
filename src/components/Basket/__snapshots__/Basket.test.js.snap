// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Basket/> component should render 1`] = `
<a
  class="header__basket"
  href="/basket"
>
  <img
    alt=""
    src="./img/shopping-cart.png"
  />
  0
</a>
`;
